### DISCUSSION TITLE
Add setting to customize severity for fixable errors

### DISCUSSION THREAD

**Discussion by OliverJAsh** [biomejs/biome-vscode#227] [Ideas]
The ESLint VS Code extension allows you to customize the severity of all fixable lint errors with the following configuration:

```json
{
  "eslint.rules.customizations": [
    { "rule": "*", "fixable": true, "severity": "info" }
  ]
}
```

It would be useful if Biome had something similar to this. In the absence of this we need to manually downgrade each rule like so:

```diff
 {
   "linter": {
     "enabled": true,
     "rules": {
       "recommended": true,
       "style": {
+        "useTemplate": "warn"
       }
     }
   }
 }
```

**Comment by ematipico** [comment]
Can you please provide a few reasons or valid use cases for this feature? 

Just because another tool has this option isn't a valid reason.

**Reply by OliverJAsh** [reply]
Sorry, you're totally right. I think these issues should describe the problem we're trying to solve:

- https://github.com/microsoft/vscode-eslint/issues/1199
- https://github.com/microsoft/vscode-eslint/issues/468

TLDR:

> As pointed out in issue #468, most ESLint shared configs use "error" as their default rule severity, but should often be displayed as warnings in VS Code since they are often less important and thus should not be mixed with TypeScript errors. In addition, an even less important (e.g. stylistic) rule in ESLint is normally (or can be) configured with severity "warn", which should be displayed as "Information" in VS Code, to be distinguishable from more important ESLint errors (displayed as VS Code warnings).

**Reply by ematipico** [reply]
v1.8 will allow you to do that: https://github.com/biomejs/biome/pull/3014

**Reply by OliverJAsh** [reply]
Thanks for engaging, but I don't understand how the PR you linked to is related. It seems to be doing something different. Perhaps I wasn't clear about the desired behaviour. If you agree, could we re-open this?

The linked PR appears to add a new `"info"` severity, but what I want is the ability to customize the severity (most likely to `warn`) of all fixable lint errors using a pattern like the VS Code example I provided, so I don't have to do it manually for each fixable lint rule.

I should have emphasized this sentence when I quoted the issue above:

> As pointed out in issue #468, most ESLint shared configs use "error" as their default rule severity, but should often be displayed as warnings in VS Code since they are often less important and thus should not be mixed with TypeScript errors.

**Reply by ematipico** [reply]
Your idea is very similar to this proposal: https://github.com/biomejs/biome/discussions/689#discussioncomment-7516172
